### PR TITLE
feat/AN-4410: no longer waiting for fetch

### PR DESCRIPTION
**Pull request by yelenderosen**
https://intezer.atlassian.net/browse/AN-4410

### CODE REVIEW

File: edr_connect/managers/analysis_manager.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 from intezer_sdk.errors import HashDoesNotExistError
 from requests import HTTPError
 
-from edr_connect.proxies.base_edr_proxy import AlertInfo
+from edr_connect.proxies.base_edr_proxy import AlertInfo, FetchCommand
```
Comment (almogch) [inline edr_connect/managers/analysis_manager.py]: new line

File: edr_connect/proxies/base_edr_proxy.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import abc
 from dataclasses import dataclass
-from typing import List
+from typing import List, Dict
```
Comment (almogch) [inline edr_connect/proxies/base_edr_proxy.py:3]: new line

**Review by almogch — APPROVED** [review]

File: edr_connect/proxies/cs_edr_proxy.py [hunk 3]
```diff
@@ -30,6 +34,17 @@ class CSOperationError(RuntimeError):
     def __init__(self):
         super().__init__(f"Crowdstrike command didn't return any status")
 
+class CSOperationIncomplete(RuntimeError):
+    def __init__(self):
+        super().__init__(f"Crowdstrike command didn't complete")
+
+
+default_retry = retry(
+    wait=wait_fixed(0.5),
```
Comment (davidt99) [inline edr_connect/proxies/cs_edr_proxy.py:43]: that's 5 seconds in total, so i think it's too low

File: edr_connect/proxies/s1_edr_proxy.py [hunk 4]
```diff
@@ -141,35 +140,32 @@ def assert_response(self, response: requests.Response):
...
+        if not fetched_alert:
+            _logger.debug('Sending fetch command to the endpoint')
+            self.post(FETCH_FILE_ROUTE,
+                      json={'data': {'password': zip_password}, 'filter': {'ids': [alert_id]}})
+            fetched_alerts[alert_id] = FetchCommand(alert_id, 0)
             return None, None
 
+        response = self.get(ACTIVITIES_ROUTE,
+                            params={'threatIds': alert_id,
+                                    'activityTypes': 86,
```
Comment (kfirstri) [inline edr_connect/proxies/s1_edr_proxy.py:156]: what is this 86 magic number?

File: edr_connect/proxies/cs_edr_proxy.py [hunk 5]
```diff
@@ -203,65 +216,85 @@ def download_quarantined_file(self, alert_info: CSAlertInfo):
...
 
-                self._wait_for_command_completion(response)
-                file = self.real_time_response.get_extracted_file_contents(session_id=session_id,
-                                                                           sha256=alert_info.file_hash,
-                                                                           filename=file_path)
-                _logger.debug(f'Download completed')
-                return file
+            _, sha256 = self._get_full_name_and_file_sha256(session_id, file_path)
+            if not sha256:
+                _logger.debug(f'File did not fetch for alert: {alert_info.alert_id}')
```
Comment (kfirstri) [inline edr_connect/proxies/cs_edr_proxy.py:282]: this debug log is the same as in the _Fetch method.. can it be confusing?

File: edr_connect/proxies/cs_edr_proxy.py [hunk 6]
```diff
@@ -154,23 +173,17 @@ def normalize_alerts_info(self, alerts_info: List[dict]) -> List[AlertInfo]:
...
-
-        err_msg = ('Time out waiting for command, this is most likely when the endpoint is powered off '
-                   'or the agent is shut down')
-        _logger.error(err_msg)
-        raise CSOperationError()
+        if not result['body']['resources']:
+            raise CSOperationError()
+
+        result = self.real_time_response.check_command_status(cloud_request_id=cloud_request_id)
+        if not result['body']['resources'][0]['complete']:
```
Comment (kfirstri) [inline edr_connect/proxies/cs_edr_proxy.py:183]: maybe deep_get here or something?

**Review by kfirstri — APPROVED** [review]

File: edr_connect/proxies/s1_edr_proxy.py (hunk 4 above)
Comment (yelenderosen) [reply edr_connect/proxies/s1_edr_proxy.py:156]: the activity type for a file upload

File: edr_connect/proxies/cs_edr_proxy.py (hunk 5 above)
Comment (yelenderosen) [reply edr_connect/proxies/cs_edr_proxy.py:282]: one is for CS, other is for S1

File: edr_connect/proxies/cs_edr_proxy.py (hunk 3 above)
Comment (yelenderosen) [reply edr_connect/proxies/cs_edr_proxy.py:43]: changed to 15 seconds

**Pull request merged**